addons/project_hr_expense/models/project.py

```diff
@@ -97,10 +97,10 @@ def _get_expenses_profitability_items(self, with_action=True):
         }
         if can_see_expense:
             args = [section_id, [('id', 'in', expense_data['ids'])]]
-            if expense_data['ids']:
-                args.append(expense_data['ids'])
+            if len(expense_data['ids']) == 1:
```

Comment:
Sorry but I don't understand your changes, you now check there is only one id? 🤔

Reply:
Hello @xavierbol, Here ```args = [section_id, [('id', 'in', expense_data['ids'])]]```, the second element is domain. If we have a single expense it is shown in form view if there are multiple expenses it is shown list view. So to get the action we pass through function ```action_profitability_items```, which in passes res_id if there is a res_id, if there are none it passes a empty list. It calls another method ```_get_expense_action``` [here](https://github.com/odoo/odoo/blob/aecdab5448b868dd35307293364c897771890afe/addons/project_hr_expense/models/project.py#L37) that the list view or multiple expense ids being handled by domain but not by res_ids only form view is being handled by res_id. Also domain contains all the res_ids.What do you think?

Reply:
Hello @kthe-odoo ,

Ah yeah indeed, you right, thanks for the explanation. 🙂